### DISCUSSION TITLE
Consider remove bendctl

### DISCUSSION THREAD

**Discussion by bohutang** [databendlabs/databend#4423] [General]
bendctl is no used now, and we recommend that use our guideline to deploy Databend:
https://databend.rs/doc/category/deploy

We need redesign the new `bendctl` in the future to meet the needs of on-perm deployment.

cc @ZhiHanZ

**Comment by sundy-li** [comment]
I think we can upgrade `bendctl` to `bendsql` or `bend-cli` as [snowsql](https://docs.snowflake.com/en/user-guide/snowsql.html)

**Comment by bohutang** [comment]
Hello @sundy-li ,

`snowsql` is not a deploy tool, for now we just need to provide a deployment tool for the community user easy to use.

**Comment by PsiACE** [comment]
I think we can keep just a basic version management and deployment tool. It looks like nvm does.

Also, we can design it like cargo to make it easy for others to load their own custom subcommands.

**Comment by ZhiHanZ** [comment]
I think the main challenge for current bendctl is the pressure from maintainence. Further adoption would be focused on on-prem k8s cluster deployment or interact with cloud warehouses. For local environment, mainly used for test and demo purpose, we could depreciate it for now
